### ISSUE TITLE
Inconsistent google_kms_crypto_key.id format on terraform apply

### ISSUE THREAD

**Issue by li0nel** [hashicorp/terraform-provider-google#11183]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/google v3.85.0
```
### Affected Resource(s)

* google_kms_crypto_key

### Terraform Configuration Files

```tf
resource "google_kms_key_ring" "key_ring" {
  name     = "debug-keyring"
  location = "australia-southeast1"
}

resource "google_kms_crypto_key" "crypto_key" {
  name            = "debug-key"
  key_ring        = google_kms_key_ring.key_ring.id
#   rotation_period = "7776000s"
}
```

### Expected Behavior

```
INFO   2022-02-24 08:43:29      ~ resource "google_kms_crypto_key" "crypto_key" {
INFO   2022-02-24 08:43:29            id                            = "projects/xxx/locations/australia-southeast1/keyRings/xxx-keyring/cryptoKeys/xxx-key"
INFO   2022-02-24 08:43:29            name                          = "xxx-key"
INFO   2022-02-24 08:43:29          - rotation_period               = "7776000s" -> null
INFO   2022-02-24 08:43:29            # (6 unchanged attributes hidden)
INFO   2022-02-24 08:43:29    
INFO   2022-02-24 08:43:29    
INFO   2022-02-24 08:43:29            # (2 unchanged blocks hidden)
INFO   2022-02-24 08:43:29        }
```
### Actual Behavior

```
INFO   2022-02-24 08:43:29      ~ resource "google_kms_crypto_key" "crypto_key" {
INFO   2022-02-24 08:43:29            id                            = "australia-southeast1/xxx-keyring/xxx-key"
INFO   2022-02-24 08:43:29            name                          = "xxx-key"
INFO   2022-02-24 08:43:29          - rotation_period               = "7776000s" -> null
INFO   2022-02-24 08:43:29            # (6 unchanged attributes hidden)
INFO   2022-02-24 08:43:29    
INFO   2022-02-24 08:43:29    
INFO   2022-02-24 08:43:29            # (2 unchanged blocks hidden)
INFO   2022-02-24 08:43:29        }
```

### Steps to Reproduce

Unfortunately I cannot reproduce locally and activate the debug traces.

On some TF updates, the ID format of `google_kms_crypto_key` is compromised into a shorter URL, which fails input validation on other resources' API calls such as `google_pubsub_topic ` which uses `kms_key_name = google_kms_crypto_key.crypto_key.id`

### References

* #0000

**Comment by shuyama1** [comment]
Hi @li0nel! I was trying to reproduce this error but it worked fine for me. I also checked how we generated the `google_kms_crypto_key` and `google_kms_key_ring ` for both resource and datasource, and it looks like that they all should be generated in the correct format. Are you still seeing this issue recently? If so, would you mind activating the debug traces and sharing the debug log here?

**Comment by shuyama1** [comment]
Okay. Looks like you are using provider version v3.85.0. That makes sense now. Sorry for missing it when reading the issue. The problem should be fixed in our 4.0.0 major release. If you upgrade the provider to any version later than 4.0.0 (the latest version is [4.12.0](https://registry.terraform.io/providers/hashicorp/google/4.12.0)), it should be working I believe.

**Comment by shuyama1** [comment]
Closing this bug! Please feel free to reopen it or file a new one if this issue happens again.
